### PR TITLE
Make coord_sf() default use projected coords

### PR DESCRIPTION
**Pull request by clauswilke**
Set `default_crs = NULL`, as discussed in #4494.

### CODE REVIEW

File: tests/figs/deps.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 - vdiffr-svg-engine: 1.0
 - vdiffr: 0.3.3
-- freetypeharfbuzz: 0.2.6
+- freetypeharfbuzz: 0.2.5
```
Comment (clauswilke) [inline tests/figs/deps.txt:3]: @thomasp85 Not entirely sure why this change happened. It doesn't seem to affect any of the reference images. If you think it matters, please let me know what I need to do to change this back. I updated to the latest vdiffr but that didn't seem to make a difference.

**Comment by clauswilke** [comment]
I think this is done, but I'd like to do a bit more testing and review before committing to these changes.

**Comment by MikkoVihtakari** [comment]
One comment I can see just by skimming the diff: just `default_crs = 4326` should be enough (the integer should be passed to st_crs internally) as with the `crs` argument in previous versions of ggplot2. Now it says "`default_crs = sf::st_crs(4326)`" in the documentation.

**Comment by clauswilke** [comment]
Yes, `default_crs = 4326` is sufficient. Just trying to be a bit more explicit and precise in the official documentation. For example, the `datum` argument has always been `datum = sf::st_crs(4326)`.

**Comment by MikkoVihtakari** [comment]
Yep, I saw that now. It's in Usage though. Minor details, but perhaps just write in Details that both `sf::st_crs(4326)` and 4326 are sufficient and that 4326 will be passed to `sf::st_crs`?

File: tests/figs/deps.txt (hunk 1 above)
Comment (thomasp85) [reply tests/figs/deps.txt:3]: I think it is because you have an older version of freetypeharfbuzz installed and it records the version last used to validate the tests

**Review by thomasp85 — APPROVED** [review]
